dask_mpi/tests/test_cli.py

```diff
@@ -25,7 +25,7 @@
 def test_basic(loop, nanny, mpirun):
     with tmpfile(extension="json") as fn:
 
-        cmd = mpirun + ["-np", "4", "dask-mpi", "--scheduler-file", fn, nanny]
+        cmd = mpirun + ["-np", "4", "dask-mpi", "--scheduler-file", fn, nanny, "&"]
 
         with popen(cmd):
             with Client(scheduler_file=fn) as c:
```

Comment:
This is the culprit line causing the tests to hang.  It is in this line that the `distributed.Client` is being initialized and told to find the Scheduler by looking up the `address` parameter in the `scheduler_file`.  The `scheduler_file` (called `fn` in this test) is not being used by `dask-mpi`, so no information is being written to the `scheduler_file` from the Scheduler.  Hence, the Client keeps waiting and doesn't know how to find and connect to the Scheduler.

It is unclear to me how this should change if the Scheduler uses an MPI broadcast to tell the Workers to what address to connect.  The Client is left out of the broadcast, and so has no means of knowing where the Scheduler is located.  (This is different in the `core.py` code, where the Client's process actually participates in the broadcast, and therefore knows where to the Scheduler is located.)

The correct solution to this problem probably requires some discussion.

Reply:
I might also note that when you run `dask-mpi` by hand, as is commonly done, then you can get the address of the Scheduler from the log messages, and you can connect your Client to that.  In this case, though, when you are automating the launch of `dask-mpi` in the test, the only way I know of getting the Scheduler address is (1) through a `scheduler_file` or (2) scrubbing STDOUT (or the log file) from the Scheduler process to find the address.  Option (1) defeats the purpose of using the MPI broadcast, and Option (2) seems messy to me.

Reply:
Clearly this decision is up to the main developers, but IMHO the unit tests should reflect how people are commonly using `dask-mpi` and not the corner cases that stem from the CI setup.

That said, I very much appreciate your help in carefully looking over this PR @kmpaul and @mrocklin !

Reply:
You are more than welcome, @lastephey!  And I very much appreciate your PR.  Help on OSS is always appreciated.  😄 

I wouldn't call this a corner case.  The difficulty in getting this test to pass with the previous code changes was a sign that something was wrong.  And the issue was that by using an MPI broadcast to send the scheduler info to the workers, there was no way to *also* send the scheduler info to the *client* (i.e., the client created in the test).  Not without using a `scheduler_file`, and the use of a `scheduler_file` seems redundant if you are using an MPI broadcast.